app/scripts/controllers/table.js

```diff
@@ -351,11 +343,53 @@ app.controller('TableCtrl', ['$location', '$scope', '$modal', 'ForamAPIService',
       });
 
       modalInstance.result.then(function (loadedFilter) {
+        $scope.loadedFilterSet._id = loadedFilter._id;
+        $scope.loadedFilterSet.name = loadedFilter.name;
         unflattenFilter(loadedFilter);
       });
     };
 
+    $scope.updateFiltersSet = function () {
+      flatFilters = prepareFilters();
```

Comment:
Nie podoba mi się to, że zmienna `flatFilters` jest definiowana w nadrzędnym bloku. Wszystkie metody traktują ją jak zmienną lokalną - na samym początku wykonują przypisanie z `prepareFilters()`, czyli od razu nadpisują to co inne metody mogły wnieść do wartości tej zmiennej, następnie dodają jakieś `name` albo `id` żeby na końcu wykonać zapytanie do API ze zmienną podaną jako argument. Skoro wszystkie metody traktują zmienną `flatFilters` jako zmienną tymczasową do komunikacji z API, nie ma sensu wprowadzać zamieszania z definicją poziom wyżej. Wykonaj definicję w bloku tej metody, np.

```javascript
var updatedFiltersSet = prepareFilters();
```